File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/daos/KardexDao.java
package mx.edu.upqroo.kristenandroid.data.database.daos;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import mx.edu.upqroo.kristenandroid.data.database.entities.Kardex;

@Dao
public interface KardexDao {
    @Insert
    long insert(Kardex kardex);

    @Update
    void update(Kardex... kardex);

    @Delete
    void delete(Kardex... kardex);

    @Query("DELETE FROM kardex")
    void deleteAll();

    @Query("SELECT * FROM kardex WHERE kardexId = :kardexId")
    LiveData<Kardex> getById(long kardexId);

    @Query("SELECT * FROM kardex")
    LiveData<List<Kardex>> getAll();

    @Query("SELECT * FROM kardex WHERE userId = :userId")
    LiveData<List<Kardex>> getByUserId(final String userId);

    @Query("SELECT COUNT(*) FROM kardex")
    int count();
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/daos/KardexDao.kt
package mx.edu.upqroo.kristenandroid.data.database.daos

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import mx.edu.upqroo.kristenandroid.data.database.entities.Kardex

@Dao
interface KardexDao {

    @get:Query("SELECT * FROM kardex")
    val all: LiveData<List<Kardex>>

    @Insert
    fun insert(kardex: Kardex): Long

    @Update
    fun update(vararg kardex: Kardex)

    @Delete
    fun delete(vararg kardex: Kardex)

    @Query("DELETE FROM kardex")
    fun deleteAll()

    @Query("SELECT * FROM kardex WHERE kardexId = :kardexId")
    fun getById(kardexId: Long): LiveData<Kardex>

    @Query("SELECT * FROM kardex WHERE userId = :userId")
    fun getByUserId(userId: String): LiveData<List<Kardex>>

    @Query("SELECT COUNT(*) FROM kardex")
    fun count(): Int
}
